starknet_py/net/client_models.py

```diff
@@ -346,6 +346,41 @@ class ContractClass:
     abi: Optional[AbiDictList] = None
 
 
+@dataclass
+class NewEntryPoint:
+    """
+    Dataclass representing contract entry point
+    """
+
+    function_idx: int
+    selector: int
+
+
+@dataclass
+class NewEntryPointsByType:
+    """
+    Dataclass representing contract class entrypoints by entry point type
+    """
+
+    constructor: List[NewEntryPoint]
+    external: List[NewEntryPoint]
+    l1_handler: List[NewEntryPoint]
+
+
+@dataclass
+class NewContractClass:
+    """
+    Dataclass representing Cairo1 contract declared to Starknet
+    """
+
+    contract_class_version: str  # Should this be added? Purpose of it is unclear to me.
+    program: dict
+    entry_points_by_type: NewEntryPointsByType
+    abi: Optional[dict] = None
+    # TODO: Better typing for this will be needed once we start Cario1 properly.
+    #  Not sure if the ABI structure is final already. It is typed as `str` in cairo-lang.
```

Comment:
A problem with this: Typechecker is complaining about the some methods in `contract_abi_resolver.py` being incompatible with `NewContractClass` and `abi` being typed as just `dict`. This will probably require some changes before this PR is merged.